contracts/Ethleverage.sol

```diff
@@ -0,0 +1,292 @@
+
+    uint recycledPeth;
+    recycledPeth = sender.prinContr;
+
+    // Step 3. Open CDPContract
+    bytes32 CDPInfo = tub.open();
+    sender.cdps = CDPInfo;
+    CDPOpened(CDPInfo); //event
+
+    // 1. convert eth into WETH
+    //just in case the MakeGuy was wrong: wethContract.transfer(recycledPeth);
+    weth.deposit.value(recycledPeth)();
+    uint DaiAmount = (_priceFloor.div(makerLR)).mul(100);
+
+    for (uint i = 0; i < sender.layers; i++) {
+        // 2a. Convert WETH into PETH
+        tub.join(recycledPeth);
+
+         // 4. deposit PETH into CDP
+        tub.lock(sender.cdps, recycledPeth);
+        PethLocked(CDPInfo, recycledPeth); //event
+
+         // 5. withdraw DAI
+        tub.draw(sender.cdps, DaiAmount); // may need to use liquidation ratio in this!
+
+        //6. buy weth, sell dai
+				//recycledPeth = IMoneyMaker(moneyMakerKovan).buyAllEthWithDai().div(1e18); //on mainNet
+        ILiquidator(liquidKovan).bust(DaiAmount); //for Kovan use only (NOTE: will not work if Total Liquidity Available is 0)
+        recycledPeth = DaiAmount.div(_ethMarketPrice);
+        dai2Peth(recycledPeth); //event
+
+        //Assign the last DaiAmount recieved in the loop
+        sender.daiAmountFinal = DaiAmount;
+
+        DaiAmount = (recycledPeth.mul(_ethMarketPrice)).div(sender.CR);
+      }
+
+    return true;
+   }
+
+   function transferOwnership(address _destination) onlyInvestor public returns (bool success) {
+     ICDPContract(CDPContract).give(investors[msg.sender].cdps, _destination);
+     return true;
+   }
+
+
+   function liquidate() onlyInvestor public returns (bool sufficient) {
+
+     Investor memory sender;
+     sender = investors[msg.sender];
+		 uint DaiAmount;
+		 uint releasedPeth;
+
+     //assign the final dai amount to the amount of dai that must be iterated
+     DaiAmount = sender.daiAmountFinal;
+
+     //1. wipe off some of the the debt by paying back some of the Dai amount
+     ICDPContract(CDPContract).wipe(sender.cdps, DaiAmount);
+
+
+     for (uint i = 0; i < sender.layers; i++) {
+
+       //2. free up some of the collatoral (PETH) because some of the debt has been wiped
+       releasedPeth = DaiAmount.mul(sender.CR);
+       ICDPContract(CDPContract).free(sender.cdps, releasedPeth);
+
+       //3. convert PETH to WETH
+       ICDPContract(CDPContract).exit(releasedPeth);
+
+       //4. trade some ethereum for DAI
+       DaiAmount = IMoneyMaker(moneyMakerKovan).sellAllEthForDai();
+
+       //5. wipe off some of the the debt by paying back some of the Dai amount
+       ICDPContract(CDPContract).wipe(sender.cdps, DaiAmount);
+     }
+
+     //6. take out the remaining peth
+     releasedPeth = DaiAmount.mul(sender.CR);
+     ICDPContract(CDPContract).free(sender.cdps, releasedPeth);
+
+     //convert PETH to WETH
+     ICDPContract(CDPContract).exit(releasedPeth);
+
+     //convert WETH to ETH
+
+     //close down the cpds
+     ICDPContract(CDPContract).shut(sender.cdps);
```

Comment:
Just for reference: when we shut a cdp, we need to pay the fees (stability & governance) in MKR. This won't be the case in the next deployment of dai (coming in the summer), but it might be reason that we'll have trouble with liquidate for now.